packages/adp-tooling/src/writer/options.ts

```diff
@@ -153,3 +164,151 @@ function getOpenSourceMiddlewares(config: AdpWriterConfig): CustomMiddleware<obj
+        });
+    }
+
+    inboundContent.entityPropertyChange.push({
+        propertyPath: "signature/parameters/sap-appvar-id",
+        operation: "UPSERT",
+        propertyValue: {
+            required: true,
+            filter: {
+                value: appId,
+                format: "plain"
+            },
+            launcherValue: {
+                value: appId
+            }
+        }
+    });
+
+    return inboundContent;
+}
+
+/**
+ * Get a Inbound change content without provided inboundId.
+ *
+ * @param flpConfiguration FLP cloud project configuration
+ * @param appId application id
+ * @returns Inbound change content.
+ */
+function getInboundChangeContentWithNewInboundID(flpConfiguration: NewInboundNavigation, appId: string): InboundChangeContentAddInboundId {
+    if (!flpConfiguration.action || !flpConfiguration.semanticObject || !flpConfiguration.inboundId) {
+        throw new Error("Missing properties!");
+    }
+
+    const content: InboundChangeContentAddInboundId = {
+        inbound: {
+            [flpConfiguration.inboundId]: {
+                action: flpConfiguration.action,
+                semanticObject: flpConfiguration.semanticObject,
+                title: `{{${appId}_sap.app.crossNavigation.inbounds.${flpConfiguration.inboundId}.title}}`,
+                signature: {
+                    additionalParameters: "allowed",
+                    parameters: flpConfiguration.additionalParameters ?? {}
+                }
+            }
+        }
+    };
+
+    if (flpConfiguration.subTitle) {
+        content.inbound[flpConfiguration.inboundId].subTitle = `{{${appId}_sap.app.crossNavigation.inbounds.${flpConfiguration.inboundId}.subTitle}}`;
+    }
+
+    content.inbound[flpConfiguration.inboundId].signature.parameters["sap-appvar-id"] = {
+        required: true,
+        filter: {
+            value: appId,
+            format: "plain"
+        },
+        launcherValue: {
+            value: appId
+        }
+    };
+
+    return content;
+}
+
+/**
+ * Generate Inbound change content required for manifest.appdescriptor.
+ *
+ * @param flpConfiguration FLP cloud project configuration
+ * @param appId application id
+ */
+export function enhanceManifestChangeContentWithFlpConfig(flpConfiguration: FlpConfig, app: CloudApp, manifestChangeContent: Content[] | Content[] = [], addInboundId: boolean): void {
+    const inboundChangeContent = addInboundId ? getInboundChangeContentWithNewInboundID(flpConfiguration as NewInboundNavigation, app.id) : getInboundChangeContentWithExistingInboundId(flpConfiguration as ChangeInboundNavigation, app.id);
+    if (inboundChangeContent) {
+        const firstFlpChange = {
+            changeType: addInboundId ? "appdescr_app_addNewInbound" : "appdescr_app_changeInbound",
+            content: inboundChangeContent,
+            texts: {
+                "i18n": "i18n/i18n.properties"
+            }
+        };
+        const secondFlpChange = {
```

Comment:
Could we have a more descriptive name than first and second change .e.g addInboundChange and removeOtherInboundsChange

Reply:
done